source/exe/process_wide.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include "common/common/cleanup.h"
+
+namespace Envoy {
+
+// Process-wide lifecycle events for global state in third-party dependencies,
+// e.g. gRPC, c-ares.
+class ProcessWide {
```

Comment:
Very mild preference to just have the `ProcessWide` constructor do the setup, and then it's destructor do the cleanup, but don't feel strongly about it if you prefer it this way.

Reply:
+1